regular_expressions_groups/regular_expressions_groups.md

```diff
@@ -0,0 +1,280 @@
+
+![The regex101 website with the aforementioned pattern and test string added.  Four groups are indicated in colorful highlights, listed in the match information pane, and explained in depth in the explanation pane.](media/capturing_groups.png)<!-- style = "border: 1px solid rgb(var(--color-highlight));" -->
+
+We see four groups listed in the "Match Information" pane on the right side.  Above that pane, the "Explanation" pane goes into detail about each group.
+
+
+### Quiz: Capturing Groups
+
+Consider a regular expression with capturing group that describes email addresses:
+
+`[A-Za-z\d\-\._]+@([A-Za-z\d\-\._]+\.[A-Za-z]{2,})`
+
+ What will the capturing group in this regular expression actually capture?
+
+[( )] The entire email address
+[( )] The user mailbox (the entire part before the `@` of the email address)
+[(X)] The email domain (the entire part following the `@` of the email address)
+[( )] The TLD, or "top level domain" (the last period and last few letters, like `.com`, `.edu`, or `.gov`)
+[( )] None of these
+[[?]] Hint: Feeling stuck?  Try using [Regex 101](https://www.regex101.com)!
+***
+<div class = "answer">
+
+This regex looks intimidating, but it's really only a few parts.  Let's break it apart considering the email address `regex_is_awesome_123@fakemail.org`
+
+* `[A-Za-z\d\-\._]+` means one or more letter, number, hyphen, period, or underscore (in our case, "regex\_is\_awesome\_123")
+* `@` indicates the `@`
+* `[A-Za-z\d\-\._]+` again means one or more letter, number, hyphen, period, or underscore ("fakemail")
+* `\.` indicates a literal period
+* `[A-Za-z]{2,}` indicates two or more letters ("org").
+
+In our case, we had a capturing group around the last three of those bullet points.  That means that what is captured is "fastmail.org", the entire email domain.
+
+</div>
+***
+
+## The Logical "Or"
+
+Sometimes you want to define two different pattern options at a certain point in a complex string.  For example, let's consider a mouse model study of the immune system's interaction with insulin.  
+
+Your wild type subjects are given identifiers that include `WT` (wild type), followed by a hyphen, and a subject identifier of usually five (but possibly 6) digits.  You additionally have PBMC humanized mice, in addition to wild type mice, and their identifiers begin with `PBMC`, followed by a hyphen, followed by 5-6 digits.
+
+There's a handy way to indicate a logical "or" in regex, using the vertical pipe (`|`).  Text on either side of a `|` character are alternative values for the given part of a pattern. 
+
+For example, `8|5` indicates that either an 8 or a 5 appears at a given location. `banana|B` indicates that the word "banana" or the abbreviation "B" appears.  
```

Comment:
```suggestion
For example, `8|5` indicates that either an 8 or a 5 appears at a given location.  Similarly,`banana|B` indicates that the word "banana" or the abbreviation "B" appears.  
```
Oooo, this is the first time I've noticed a "sentences start with words not symbols" example :)